shared/directives/auto-focus/auto-focus.directive.ts

```diff
@@ -0,0 +1,15 @@
+import { AfterViewInit, Directive, ElementRef } from '@angular/core';
+
+@Directive({
+  selector: '[appAutoFocus]',
+})
+export class AutoFocusDirective implements AfterViewInit {
+  constructor(
+    private elementRef: ElementRef,
```

Comment:
specify some `<Type there>`, so nativeElement won't have `any` type and call focus safely with typescript